bigframes/session/metrics.py

```diff
@@ -40,8 +40,11 @@ def count_job_stats(
     ):
         if query_job is None:
             assert row_iterator is not None
-            if (row_iterator.total_bytes_processed is None) or (
-                row_iterator.query is None
+            if (
+                not hasattr(row_iterator, "total_bytes_processed")
+                or not hasattr(row_iterator, "query")
+                or (row_iterator.total_bytes_processed is None)
+                or (row_iterator.query is None)
```

Comment:
The object shouldn't be mutated while we're iterating over it but we can be a bit more robust by using `getattr` instead.

```suggestion
                (total_bytes_processed := getattr(row_iterator, "total_bytes_processed", None)) is None
                (query := getattr(row_iterator, "query", None)) is None
```